test/tools/cli/IonProcess.java

```diff
@@ -183,79 +210,411 @@ private static void processEvent(IonWriter ionWriterForOutput,
+        SymbolToken field = event.getFieldName();
+        if (field != null) {
+            if (field.getText() != null) {
+                ionWriter.setFieldNameSymbol(field);
+            } else {
+                ionWriter.setFieldName("$0");
+            }
+        } else {
+            if (ionWriter.isInStruct()) ionWriter.setFieldName("$0");
+        }
+
+        switch (ionType) {
+            case NULL:
+                ionWriter.writeNull();
+                break;
+            case BOOL:
+                ionWriter.writeBool(Boolean.parseBoolean(event.getValueText()));
+                break;
+            case INT:
+                ionWriter.writeInt(Integer.parseInt(event.getValueText()));
+                break;
+            case FLOAT:
+            case DECIMAL:
+                ionWriter.writeFloat(Double.parseDouble(event.getValueText()));
+                break;
+            case TIMESTAMP:
+                ionWriter.writeTimestamp(Timestamp.valueOf(event.getValueText()));
+                break;
+            case SYMBOL:
+                ionWriter.writeSymbol(event.getValueText());
+                break;
+            case STRING:
+                String string = event.getValueText();
+                String subString = string.substring(1, string.length()-1);
+                ionWriter.writeString(subString);
+                break;
+            case CLOB:
+                ionWriter.writeClob(event.getValueBinary());
+                break;
+            case BLOB:
+                ionWriter.writeBlob(event.getValueBinary());
+                break;
+            default:
+                throw new IonException("invalid ion_type " + ionType.toString());
+        }
+    }
+
+    private static Event ionStreamToEvent(IonReader ionReader) {
+        Event event = new Event();
+        ionReader.stepIn();
+        while (ionReader.next() != null) {
+            switch (ionReader.getFieldName()) {
+                case "event_type":
+                    event.setEventType(EventType.valueOf(ionReader.stringValue().toUpperCase()));
+                    break;
+                case "ion_type":
+                    event.setIonType(IonType.valueOf(ionReader.stringValue().toUpperCase()));
+                    break;
+                case "field_name":
+                    ionReader.stepIn();
+                    String fieldText = null;
+                    int fieldSid = -1;
+                    while (ionReader.next() != null) {
+                        switch (ionReader.getFieldName()) {
+                            case "text":
+                                fieldText = ionReader.stringValue();
+                                break;
+                            case "sid":
+                                fieldSid = ionReader.intValue();
+                                break;
+                        }
+                    }
+                    FakeSymbolToken symbolToken = new FakeSymbolToken(fieldText, fieldSid);
```

Comment:
Changed,

If I use _Private_Utils.newSymbolToken(fieldText, fieldSid); the FieldText cant be null.
For example, there is an event stream whose fieldName is empty. For Ion-c it will print $0 but for ion-java, we can't create a null text symbolToken so it will throw an error. Is that expected?

Reply:
I don't see anything in `_Private_Utils.newSymbolToken(fieldText, fieldSid)` that prevents you from providing `null` fieldText. Where did you find that?

Reply:
Got it, fieldText must be a string rather than null value.

For example _Private_Utils.newSymbolToken(null, 0) is invalid but 
For example _Private_Utils.newSymbolToken(a_null_string, 0)  is valid.